### DISCUSSION TITLE
KeystoneJS GraphQL Errors in not compatible with specification

### DISCUSSION THREAD

**Discussion by pahaz** [keystonejs/keystone#5913] [Feature requests]
I have a Message list with a virtual field `sender` with custom validation. 

If we look at my query:
```
mutation {
    obj: createMessage (data: ...) {
        id
        sender
    }
}
```

After I run the query I will get an error:
```
{
    ...,
    errors: [{
        name: 'ValidationFailureError',
        message: 'You attempted to perform an invalid mutation',
        path: ['obj'],
        time_thrown: ...
        data: { messages: ['real error message'], listKey: 'Message', operation: 'create', errors: [{}] }
    }]
}
```

But if we look at the GraphQL http://spec.graphql.org/draft/#sec-Errors we can expect something like:
```
{
    ...,
    errors: [{
        message: 'real error message',
        path: ['obj', 'sender'],
        extensions: {
            code: 'BAD_USER_INPUT',
            time_thrown: ...
            listKey: 'Message', 
            operation: 'create',
        },
    }]
}
```

Here you can use `path` and display errors near the form field (related to https://github.com/keystonejs/keystone/discussions/5881). 
I've also used `code` from Apollo errors: https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes 

Currently, it's a little hard to process the KeystoneJS errors because it used `apollo-errors` library which has not been updated for more than two years. And the Apollo Server already has good error processing support.

**Comment by bladey** [comment]
Hi @pahaz, we know our errors aren’t ideal right now, reviewing our error structures is something that’s on our roadmap and that we’re hoping to address real soon.
